framework/include/transfers/MultiAppCopyTransfer.h

```diff
@@ -9,56 +9,36 @@
 
 #pragma once
 
-#include "MultiAppTransfer.h"
+#include "MultiAppFieldTransfer.h"
 
 // Forward declarations
 class MultiAppCopyTransfer;
-class MooseVariableFEBase;
-namespace libMesh
-{
-class DofObject;
-}
 
 template <>
 InputParameters validParams<MultiAppCopyTransfer>();
 
 /**
  * Copy the value to the target domain from the nearest node in the source domain.
  */
-class MultiAppCopyTransfer : public MultiAppTransfer
+class MultiAppCopyTransfer : public MultiAppFieldTransfer
 {
 public:
   static InputParameters validParams();
 
   MultiAppCopyTransfer(const InputParameters & parameters);
 
-  /**
-   * Performs basic error checking that the variable exists on MultiApp.
-   */
-  virtual void initialSetup() override;
-
   /**
    * Performs the transfer of a variable (Nonlinear or Auxiliary) to/from the Multiapp.
    */
   virtual void execute() override;
 
 protected:
-  /**
-   * Performs the transfer of a variable between two problems.
-   */
-  void transfer(FEProblemBase & to_problem, FEProblemBase & from_problem);
-
-  /**
-   * Performs the transfer of values between a node or element.
-   */
-  void transferDofObject(libMesh::DofObject * to_object,
-                         libMesh::DofObject * from_object,
-                         MooseVariableFEBase & to_var,
-                         MooseVariableFEBase & from_var);
+  virtual std::vector<VariableName> getFromVarNames() const override { return _from_var_names; }
+  virtual std::vector<AuxVariableName> getToVarNames() const override { return _to_var_names; }
 
-  /// The name of the variable to transfer to
-  const VariableName & _to_var_name;
+  std::vector<VariableName> _from_var_names;
+  std::vector<AuxVariableName> _to_var_names;
 
-  /// Name of variable transfering from
-  const VariableName & _from_var_name;
+  VariableName _from_var_name;
+  AuxVariableName _to_var_name;
```

Comment:
We lost the const references to both of these members, but the bigger issue is the type change. Why did the "to" variable get changed to Aux?

Reply:
There is actually no reason these can't be const now.